Text field block loader properly handles null values from delegate (#127525)
elastic/elasticsearch: docs/reference/elasticsearch/mapping-reference/text.md

::::

> `text` fields may use a [`keyword`](/reference/elasticsearch/mapping-reference/keyword.md#keyword-synthetic-source) sub-field to support [synthetic `_source`](/reference/elasticsearch/mapping-reference/mapping-source-field.md#synthetic-source) without storing values of the text field itself.

Reviewer: ```suggestion
`text` fields can use a [`keyword`](/reference/elasticsearch/mapping-reference/keyword.md#keyword-synthetic-source) sub-field to support [synthetic `_source`](/reference/elasticsearch/mapping-reference/mapping-source-field.md#synthetic-source) without storing values of the text field itself.
```

> ::::{note}
> Synthetic source of the `text` field will have the same [modifications](/reference/elasticsearch/mapping-reference/mapping-source-field.md#synthetic-source) as a `keyword` field in this case.
> 
> These modifications can impact usage of `text` fields:
> * Reordering text fields can have an effect on [phrase](/reference/query-languages/query-dsl/query-dsl-match-query-phrase.md) and [span](/reference/query-languages/query-dsl/span-queries.md) queries. See the discussion about [`position_increment_gap`](/reference/elasticsearch/mapping-reference/position-increment-gap.md) for more detail. You can avoid this by making sure the `slop` parameter on the phrase queries is lower than the `position_increment_gap`. This is the default.
> * Handling of `null` values is different. `text` fields ignore `null` values but `keyword` fields support replacing `null`s with a value specified in the `null_value` parameter. This replacement will be represented in synthetic source.
> ::::

Reviewer: you've reframed the entire section to be about `keyword`, so I'd take this out of a note

```suggestion
In this case, the synthetic source of the `text` field will have the same [modifications](/reference/elasticsearch/mapping-reference/mapping-source-field.md#synthetic-source) as a `keyword` field.

These modifications can impact usage of `text` fields:
* Reordering text fields can have an effect on [phrase](/reference/query-languages/query-dsl/query-dsl-match-query-phrase.md) and [span](/reference/query-languages/query-dsl/span-queries.md) queries. See the discussion about [`position_increment_gap`](/reference/elasticsearch/mapping-reference/position-increment-gap.md) for more details. You can avoid this by making sure the `slop` parameter on the phrase queries is lower than the `position_increment_gap`. This is the default.
* Handling of `null` values is different. `text` fields ignore `null` values, but `keyword` fields support replacing `null` values with a value specified in the `null_value` parameter. This replacement is represented in synthetic source.
```

Reviewer: imo, it would be good to keep an example here because it's the actual synthetic part. not critical, but the `store` example is less important I think

Author: I didn't want to repeat the same example as `keyword` has. Do you think it's helpful?

Reviewer: I don't think it's intuitive for people to look for the "core" example on the keyword page ... they'd likely just scroll down to the synthetics section and copy whatever is there/assume whatever example is there represents the happy path. you could consider using a [snippet](https://elastic.github.io/docs-builder/syntax/file_inclusion/) to avoid maintaining two examples. 

I'm also ok with this merging as is if you're not super worried about it 🤷

Author: Given that we don't actually have an example with a multi-field i'll go ahead and add it. Thanks for the feedback.

> If the `text` field sets `store` to `true` then the sub-field is not used and modifications mentioned above do not apply.

Reviewer: ```suggestion
If the `text` field sets `store` to `true` then the sub-field is not used and no modifications are applied.
```
